### PR TITLE
Keg-CLI - mono-repo / cli-utils

### PR DESCRIPTION
**Pull request by lancetipton**
### Context
* Currently when ever we setup custom tasks for a tap, there's a lot of boilerplate code required
* Most of boilerplate code is duplicated between taps with custom tasks
* We need to create a separate repo where this code can live to allow reusing it, instead of duplicating it

### Updates
* Added a `repos` folder to the keg-cli
* Migrated the `keg-cli-utils` repo into the repos folder
  * Been working on this repo for a little bit here => https://github.com/lancetipton/keg-cli-utils
  * I literally just cloned it into the `repos` folder

### Notes
* As of right now, there's some code duplication between the `keg-cli/src`, and this new `repos/cli-utils`  folder
* Overtime the `keg-cli/src` code will be migrated the `repos/cli-core` folder and updated to use `repos/cli-utils` 
* There are also plans to extract more of the code from the `keg-cli/src` folder into separate repos within the `repos` folder
  * The main focus will be the `keg-cli/src/libs` folder, but potentially other parts will also be migrated as needed

### Tests
* There are no code changes with this PR, so nothing really to test.
* If you want, you can run the tests for the `repos/cli-utils`
  * It's currently being used in a [non-published tap](https://github.com/simpleviewinc/tap-visitapps-app/tree/master/tasks)
  * But it's sym-linked from my local version and not actually installed from NPM 
* To run the tests for the `cli-utils` repo follow these steps 
  * Run cmd => `keg cli && keg pr 56 && cd repos/cli-utils`  
  * Run cmd => `yarn install`
  * Run cmd => `yarn test`

### CODE REVIEW

**Review by mpcarolin — COMMENTED** [review]
Left a few comments. Let me know what you think

File: repos/cli-utils/src/task/findTask.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * @param {Array} options - Task options that can be shared across tasks
+ *
+ * @example
+ * findTask({...task definitions}, [...options])
+ *
+ * @returns {void}
+ */
+const findTask = (tasks, opts = noOpArr) => {
+  const options = [...opts]
+  const taskName = options.shift()
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/cli-utils/src/task/findTask.js:63]: Hmm, it's not modifying the input array. Line 62 right above makes a copy of it.

File: repos/cli-utils/src/runTask.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+// Check if the parent module ( task module ) has a parent
+// If it does, then it was called by the Keg-CLI
+// So we should return the task definition instead of running the task action
+module.parent
+  ? (module.exports = { runTask })
+  : (async () => {
+      const response = await runTask()
+      return response
+    })()
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/cli-utils/src/runTask.js:60]: updated

File: repos/cli-utils/src/task/findTask.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ */
+const getTaskAlias = (task, tasks) => {
+  return Object.entries(tasks)
+  .reduce((foundTask, [ name, definition ]) => {
+    return !foundTask && isArr(definition.alias)
+      ? definition.alias.includes(task)
+          ? definition
+          : foundTask
+      : foundTask
+  }, false)
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/cli-utils/src/task/findTask.js:24]: updated

File: repos/cli-utils/src/task/findTask.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ * @private
+ * @param {Object} task - Custom Task Definition
+ * @param {Array} options - Task options that can be shared across tasks
+ *
+ * @example
+ * findTask({...task definition}, [...options])
+ *
+ * @returns {Object} - Found task definition by name
+ */
+const loopTasks = (task, options) => {
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply repos/cli-utils/src/task/findTask.js:39]: updated

**Review by mpcarolin — APPROVED** [review]
Looks good, thanks 👍

**Pull request merged**